drivers/ethernet/eth_renesas_ra.c

```diff
@@ -0,0 +1,415 @@
+{
+	struct renesas_ra_eth_context *ctx = dev->data;
+	ether_extended_cfg_t *p_ether_extended_cfg =
+		(ether_extended_cfg_t *)ctx->ctrl.p_ether_cfg->p_extend;
+	/* Initialize the transmit and receive descriptor */
+	memset(p_ether_extended_cfg->p_rx_descriptors, 0x00,
+	       sizeof(ether_instance_descriptor_t) * ctx->ctrl.p_ether_cfg->num_rx_descriptors);
+	memset(p_ether_extended_cfg->p_tx_descriptors, 0x00,
+	       sizeof(ether_instance_descriptor_t) * ctx->ctrl.p_ether_cfg->num_tx_descriptors);
+
+	ether_init_buffers(&ctx->ctrl);
+}
+
+static void phy_link_state_changed(const struct device *pdev, struct phy_link_state *state,
+				   void *user_data)
+{
+	const struct device *dev = (struct device *)user_data;
+	struct renesas_ra_eth_context *ctx = dev->data;
+	R_ETHERC0_Type *p_reg_etherc;
+
+	p_reg_etherc = (R_ETHERC0_Type *)ctx->ctrl.p_reg_etherc;
+
+	ARG_UNUSED(pdev);
+
+	if (state->is_up) {
+
+		ctx->ctrl.link_change = ETHER_LINK_CHANGE_LINK_UP;
+		ctx->ctrl.previous_link_status = ETHER_PREVIOUS_LINK_STATUS_UP;
+
+		renesas_ra_eth_buffer_init(dev);
+
+		/*
+		 * ETHERC and EDMAC are set after ETHERC and EDMAC are reset in software
+		 * and sending and receiving is permitted.
+		 */
+		ether_configure_mac(&ctx->ctrl, ctx->ctrl.p_ether_cfg->p_mac_address, 0);
+
+		switch (state->speed) {
+		/* Half duplex link */
+		case LINK_HALF_100BASE_T: {
+			ctx->ctrl.link_speed_duplex = ETHER_PHY_LINK_SPEED_100H;
+			break;
+		}
+
+		case LINK_HALF_10BASE_T: {
+			ctx->ctrl.link_speed_duplex = ETHER_PHY_LINK_SPEED_10H;
+			break;
+		}
+
+		/* Full duplex link */
+		case LINK_FULL_100BASE_T: {
+			ctx->ctrl.link_speed_duplex = ETHER_PHY_LINK_SPEED_100F;
+			break;
+		}
+
+		case LINK_FULL_10BASE_T: {
+			ctx->ctrl.link_speed_duplex = ETHER_PHY_LINK_SPEED_10F;
+			break;
+		}
+
+		default: {
+			ctx->ctrl.link_speed_duplex = ETHER_PHY_LINK_SPEED_100F;
+			break;
+		}
+		}
+
+		ether_do_link(&ctx->ctrl, 0);
+		ctx->ctrl.link_change = ETHER_LINK_CHANGE_LINK_UP;
+		ctx->ctrl.previous_link_status = ETHER_PREVIOUS_LINK_STATUS_UP;
+		ctx->ctrl.link_establish_status = ETHER_LINK_ESTABLISH_STATUS_UP;
+		LOG_DBG("Link up");
+		net_eth_carrier_on(ctx->iface);
+	} else {
+		ctx->ctrl.link_change = ETHER_LINK_CHANGE_LINK_DOWN;
+		ctx->ctrl.previous_link_status = ETHER_PREVIOUS_LINK_STATUS_DOWN;
+		ctx->ctrl.link_establish_status = ETHER_LINK_ESTABLISH_STATUS_DOWN;
+		net_eth_carrier_off(ctx->iface);
+	}
+}
+
+static void renesas_ra_eth_initialize(struct net_if *iface)
+{
+	fsp_err_t err;
+	const struct device *dev = net_if_get_device(iface);
+	struct renesas_ra_eth_context *ctx = dev->data;
+	const struct renesas_ra_eth_config *cfg = dev->config;
+
+	LOG_DBG("eth_initialize");
+
+	net_if_set_link_addr(iface, ctx->mac, sizeof(ctx->mac), NET_LINK_ETHERNET);
+
+	if (ctx->iface == NULL) {
+		ctx->iface = iface;
+	}
+
+	ethernet_init(iface);
+
+	const char *phy_connection_type = DT_INST_PROP_OR(0, phy_connection_type, "rmii");
```

Comment:
Please introduce variables at the start of the func

Reply:
I moved this to device init function